src/main/java/com/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java

```diff
@@ -0,0 +1,120 @@
+    public Product update(Product item) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection
+                        .prepareStatement(UPDATE_SQL, Statement.RETURN_GENERATED_KEYS)) {
+            setValues(preparedStatement, item);
+            preparedStatement.setLong(3, item.getId());
+            if (preparedStatement.executeUpdate() > 0) {
+                return item;
+            } else {
+                throw new DataProcessingException("Can't update product");
+            }
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Can't update product", ex);
+        }
+    }
+
+    @Override
+    public Optional<Product> get(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection
+                        .prepareStatement(SELECT_ONE_SQL, Statement.RETURN_GENERATED_KEYS)) {
+            preparedStatement.setLong(1, id);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            if (resultSet.next()) {
+                return Optional.of(conversionToProduct(resultSet));
+            } else {
+                throw new DataProcessingException("Can't get product");
+            }
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Can't get product", ex);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection
+                        .prepareStatement(DELETE_SQL, Statement.RETURN_GENERATED_KEYS)) {
+            preparedStatement.setLong(1, id);
+            return preparedStatement.executeUpdate() > 0;
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Can't delete product", ex);
+        }
+    }
+
+    @Override
+    public List<Product> getAll() {
+        List<Product> products = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection
+                        .prepareStatement(SELECT_ALL_SQL, Statement.RETURN_GENERATED_KEYS)) {
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                products.add(conversionToProduct(resultSet));
+            }
+            return products;
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Can't delete product", ex);
+        }
+    }
+
+    private void setValues(PreparedStatement ps, Product product) throws SQLException {
+        ps.setString(1, product.getName());
+        ps.setDouble(2, product.getPrice());
+    }
+
+    private Product conversionToProduct(ResultSet resultSet) throws SQLException {
+        long id = resultSet.getLong("product_id");
```

Comment:
why do we have inconsistent naming 
product_id
productName

Reply:
done